CMakeLists.txt

```diff
@@ -813,6 +813,9 @@ if(swift_build_android AND NOT "${SWIFT_ANDROID_NDK_PATH}" STREQUAL "")
   endif()
   set(SWIFT_ANDROID_PREBUILT_PATH
       "${SWIFT_ANDROID_NDK_PATH}/toolchains/arm-linux-androideabi-${SWIFT_ANDROID_NDK_GCC_VERSION}/prebuilt/${_swift_android_prebuilt_suffix}")
+  get_filename_component(CMAKE_LINKER_NAME "${CMAKE_LINKER}" NAME)
```

Comment:
I thought it was better to do this rather than checking for `SWIFT_USE_GOLD_LINKER`, but I agree it is misleading. I can replace it for the check on the macro (ld and ld.gold are the only two linkers in the ndk toolchain anyway) or add a comment.